R/qenv-eval_code.R

```diff
@@ -100,7 +103,7 @@ setMethod("eval_code", signature = c("qenv", "expression"), function(object, cod
   if (length(srcref)) {
     eval_code(object, code = paste(attr(code, "wholeSrcref"), collapse = "\n"))
   } else {
-    eval_code(object, code = paste(lang2calls(code), collapse = "\n"))
+    Reduce(function(object, x) eval_code(object, x), init = object, x = code)
```

Comment:
Is there a reason why an anonymous function is needed? I think `Reduce` can be simplified as follows, as tests pass:

```suggestion
    Reduce(eval_code init = object, x = code)
```

I am not that familiar with Reduce but this seems to work:

```
q <- within(qenv(), {i <- iris[1:5, ]})
Reduce(eval_code, q, x = c("x <- nrow(i[seq(2), ])", "x <- nrow(i[seq(3), ])"), accumulate = TRUE)
```

Reply:
It should work. I'll try